### ISSUE TITLE
easy wallet for linux/windows ?

### ISSUE THREAD

**Issue by lanaiscool** [forknote/forknote#56]
is this on the agenda ?

**Comment by nskuijpers** [comment]
Yea is there a qui wallet what I can use for my fork coin?

**Comment by theheartofraven** [comment]
I'm looking for it, too. If someone suceed to create a gui wallet, please, let me know. I'll give a beer, or two. :)

**Comment by lanaiscool** [comment]
I am going to make a web wallet but the problem is that its going to be a lot of messy code since there is no php library available for it cause it uses named parameters

**Comment by lanaiscool** [comment]
im going to try and edit existing qt wallets

**Comment by nskuijpers** [comment]
I going to build a nodeJS application for the web wallet and use simplewallet to interact or do people now a better why to connect with the rpc server?

**Comment by lanaiscool** [comment]
if that works and you get good responds however you could also just use curl post requests
thats what im going to use

**Comment by nskuijpers** [comment]
do you have a list with possible command you can post with curl?
That sounds a lot better than use simplewallet

**Comment by lanaiscool** [comment]
yeah sure you can use a command like this  curl -X POST --data '{"jsonrpc":"2.0","method":"getStatus","params":{},"id":1}‌​' http://localhost:8070/json_rpc
you need to have walletd running you can create curl request using php a list of commands is specified here: https://wiki.bytecoin.org/wiki/Bytecoin_RPC_Wallet_JSON_RPC_API

**Comment by nskuijpers** [comment]
This is what I need thankyou very much when I fisinshed with the project your the first person who get the web wallet!

**Comment by lanaiscool** [comment]
Thanks for that! i have not a lot of free time so i have only like 10% of the web wallet and almost no time to work on it also the web wallet i am working on is not correct and i need to start from scratch also to help you a little when passing params you need to do like so if want to check the balance of an address: 

`curl -X POST --data '{"jsonrpc":"2.0","method":"getBalance","params":{"address":"2785gGLpyMJhw4JFVVd4vMDeRMmvTNda9MMgQNBd8se2a1mNRcKYubnaCk5zubLSefZAAmjDk9Fyejb2hVDRZQ23MB2BgUW"},"id":1}‌​' http://localhost:8070/json_rpc`

**Comment by lanaiscool** [comment]
Do you have an email address also i got an php code to execute curl requests

`<?php
$data = array("method" => "getBalance", "params" => array('address' => 'S6E94HJzw7u831nbnY2MGSJJ7XSNCh7Sh5rArxAQd5AB4YnJ23qaB7P9F8BzrB6XAbDorv1aub6QphWxhC5stDUb2YSbXjTF5'));                                     $
$data_string = json_encode($data);

$ch = curl_init('http://localhost:8070/json_rpc');
curl_setopt($ch, CURLOPT_CUSTOMREQUEST, "POST");
curl_setopt($ch, CURLOPT_POSTFIELDS, $data_string);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_HTTPHEADER, array(
    'Content-Type: application/json',
    'Content-Length: ' . strlen($data_string))
);

$result = curl_exec($ch);
echo $result;
?>
`

This will output the balance of an address but since the address is not in your walletd container it will return invalid address

**Comment by nskuijpers** [comment]
I try to use your command "curl -X POST --data '{"jsonrpc":"2.0","method":"getBalance","params":{"address":"2785gGLpyMJhw4JFVVd4vMDeRMmvTNda9MMgQNBd8se2a1mNRcKYubnaCk5zubLSefZAAmjDk9Fyejb2hVDRZQ23MB2BgUW"},"id":1}‌​' http://localhost:8070/json_rpc" and changed the address, but I get an error.

Command:
curl -X POST --data '{"jsonrpc":"2.0","method":"getBalance","params":{"address":"a2207ab6c8797333fa61169f522227f6cf5503c99ab26121f76d342b47470b13"},"id":1}' http://127.0.0.1:38985/json_rpc

result:
{"error":{"code":-32601,"message":"Method not found"},"id":1,"jsonrpc":"2.0"}%

**Comment by prashantprabhakar** [comment]
@nskuijpers Were you able to  successfully create a web wallet? I need to create one for a forknote based coin. Any source?

**Comment by nskuijpers** [comment]
@prashantprabhakar I have the information to build it and ask people from my team to create it, but they are not started yet and we don't share it with the forknode community sorry.

**Comment by NEO2756** [comment]
@lanaiscool @nskuijpers I am getting "Levin signature mismatch" on my walletd (ran it with --local option) when I am doing [curl -X POST -i -H "Accept: application/json" -d '{"params": {},"jsonrpc": "2.0", "id": "test","method":"reset"}' http://localhost:27337/json_rpc] request.

anybody faced this issue?

**Comment by nskuijpers** [comment]
@NEO2756 don't run the daemon with --local you don't deed that to connect with walletd on localhost.

**Comment by NEO2756** [comment]
@nskuijpers thanks, walletd is connected to daemon now.
but for each request mentioned here http://forknote.net/documentation/rpc-wallet/rpc-wallet-curl-examples/ , I am geeting JOSN reply as "Method not Found" 

(eg: HTTP/1.1 200 OK
Content-Length: 82
Content-Type: application/json
Server: CryptoNote-based HTTP server

{"error":{"code":-32601,"message":"Method not found"},"id":"test","jsonrpc":"2.0"})

**Comment by nskuijpers** [comment]
You need to change the id "test" to a number.

**Comment by NEO2756** [comment]
No success with number instead of "test".
PS: JSON RPC 2.0 says id could be string, number (http://www.jsonrpc.org/specification).

**Comment by nskuijpers** [comment]
mmm maybe you miss something in the json format.

try this one: 
curl -X POST --data '{"jsonrpc":"2.0","method":"createAddress"},"id":1}' http://127.0.0.1:8050/json_rpc

**Comment by VluCash** [comment]
@nskuijpers  Can I buy the web wallet code? Send me a email

**Comment by nskuijpers** [comment]
@VluCash we don't sell the code.

**Comment by VluCash** [comment]
@nskuijepers oh ok I would like to see how it came out when it’s done.  I’m
Looking to have a web wallet for my coin. 

https://vlucash.org/downloads/

**Comment by lanaiscool** [comment]
I will put a video on youtube about the progress on the web wallet as im close to finishing the functions withdrawal should work however i made a new coin using a new tx hash so im right now mining until the balance goes to available instead of locked

**Comment by lanaiscool** [comment]
web wallet is working :) gonna make a video and it will be online in half an hour will post link here

**Comment by lanaiscool** [comment]
porgress: https://www.youtube.com/watch?v=8QqHvsTJF-g
TODO: make conversions as you see in the video instead of 712 it shows 712000000 this is because of the way the input has to be for walletd so i need to make a function that when you type 712 and click on withdraw it wil send 712000000 to the server 
For the rest i am also adding a nonimity field and fee field

**Comment by NEO2756** [comment]
Thats great!! 
One question: are you generating multiple adresses contained in one container file? If yes, how?
TIA

**Comment by nskuijpers** [comment]
That is easy with the create method from walletd.

**Comment by lanaiscool** [comment]
Like nskuijpers suggests you can just use the createAddress command

**Comment by NEO2756** [comment]
Yeah. Thanks.

**Comment by gogsard** [comment]
@lanaiscool is your web wallet keep users private key ? 

I'm working a web wallet too, trying to find to to mathematically generate the key pairs and broadcast the transactions.

**Comment by lanaiscool** [comment]
@gogsard yes it does it keeps the users private keys

**Comment by lanaiscool** [comment]
will clean up the code than release the web wallet the web wallet works for all forknote coins only thing you have to is make walletd running and block external access to the wallet d port since there is no rpc username and password.

**Comment by streamcoindev** [comment]
https://github.com/streamcoindev/ForkNote-webwallet
